content/en/blog/_posts/2021-12-15-pod-security-admission-psa-beta.md

```diff
@@ -0,0 +1,620 @@
+to default enabled ones (NamespaceLifecycle, LimitRanger,
+ServiceAccount, TaintNodesByCondition, PodSecurity, Priority,
+DefaultTolerationSeconds, DefaultStorageClass,
+StorageObjectInUseProtection, PersistentVolumeClaimResize,
+RuntimeClass, CertificateApproval, CertificateSigning,
+CertificateSubjectRestriction, DefaultIngressClass,
+MutatingAdmissionWebhook, ValidatingAdmissionWebhook,
+ResourceQuota).
+...
+```
+
+`PodSecurity` is listed in the group of default enabled admission plugins.
+
+### Configure Pod Security Admission
+
+Policies are applied to a namespace via labels. These labels are as follows:
+   * `pod-security.kubernetes.io/<MODE>: <LEVEL>` (required)
+   * `pod-security.kubernetes.io/<MODE>-version: <VERSION>` (*optional*, defaults to latest)
+
+A specific version can be supplied for each enforcement mode. The version pins the policy to the version that was defined at that Kubernetes version. This is to increase flexibility for future  Kubernetes enhancements and unknown security issues.
+
+From earlier, the modes are:
+   * `enforce` — Any Pods that violate the policy will be rejected
+   * `audit` — Pods with violations will be allowed and an audit annotation will be added
+   * `warn` — Pods that violate the policy will be allowed and a warning message will be sent back to the user.
+
+> Note: The default for enforce/warn/audit is `privileged` if not specified
+
+When to `warn`?
+
+Don't use`warn` for the exact same level+version of the policy as `enforce`. In the admission sequence, if `enforce` fails, the entire sequence fails before evaluating the `warn`.
+
+The typical uses for `warn` are:
+
+* `warn` at the same level but a different version (e.g. pin `enforce` to *restricted+v1.23* and `warn` at *restricted+latest*)
+* `warn` at a stricter level (e.g. `enforce` baseline, `warn` restricted)
+
+
+Let's create 3 namespaces and apply all 3 PSA security profiles. First, create the namespaces:
+
+```shell
+kubectl create namespace kube-test-privileged
+kubectl create namespace kube-test-baseline
+kubectl create namespace kube-test-restricted
+```
+
+Add the respective labels.
+
+```shell
+# kube-test-privileged enforces a "privileged" security policy and warns / audits on baseline
+kubectl label --overwrite ns kube-test-privileged \
+  pod-security.kubernetes.io/enforce=privileged \
+  pod-security.kubernetes.io/warn=baseline \
+  pod-security.kubernetes.io/audit=baseline
+
+# kube-test-baseline enforces a "baseline" security policy and warns / audits on restricted
+kubectl label --overwrite ns kube-test-baseline \
+  pod-security.kubernetes.io/enforce=baseline \
+  pod-security.kubernetes.io/warn=restricted \
+  pod-security.kubernetes.io/audit=restricted
+
+# kube-test-restricted enforces a "restricted" security policy and audits on restricted
+kubectl label --overwrite ns kube-test-restricted \
+  pod-security.kubernetes.io/enforce=restricted \
+  pod-security.kubernetes.io/audit=restricted
+```
```

Comment:
Consider creating the namespace before the example that uses them (or even reusing & relabeling a single namespace). That way the reader doesn't need to check back here to remember how each namespace was configured.